decidim-core/lib/decidim/commands/create_resource.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Commands
+    class CreateResource < ::Decidim::Command
+      include Decidim::Commands::ResourceHandler
+
+      # Initializes the command.
+      # @param form [Decidim::Form] the form object to create the resource.
+      def initialize(form)
+        @form = form
+      end
+
+      # Creates the result if valid.
+      #
+      # Broadcasts :ok if successful, :invalid otherwise.
+      def call
+        return broadcast(:invalid) if invalid?
+
+        transaction do
+          run_before_hooks
+          create_resource
+          run_after_hooks
+        end
+
+        broadcast(:ok, resource)
+      rescue Decidim::Commands::HookError
+        broadcast(:invalid)
+      end
+
+      protected
+
+      # @param soft [Boolean] whether to soft-create the resource or not.
+      # @usage
+      #  create_resource(soft: true) - Will soft-create the resource, returning any validation errors.
+      #  create_resource - Will create the resource, raising any validation errors.
+      def create_resource(soft: false)
+        @resource = Decidim.traceability.send(soft ? :create : :create!,
+                                              resource_class,
+                                              form.current_user,
```

Comment:
I've just remember about https://github.com/decidim/decidim/issues/10199 - maybe we should implement that delegation here?

Reply:
Done in https://github.com/decidim/decidim/commit/94771d78819c3002354367f20f75b28d814444b9